### PR TITLE
Adding stats functions and unit tests

### PR DESCRIPTION
**Pull request by Rambatino**
This Code and the repo majority, will supply the Crosstab and the Zappi Stats with combinatorial statistics. With reference to https://github.com/scipy/scipy/issues/9860, it is unlikely that scipy will accept recursive statistics and thus this repo will provide methods for enabling that.

### CODE REVIEW

**Comment by xulaus** [comment]
If this is the copy paste it lookslike then this is probably ok. Do you know if we are going to go submodule or package?

**Comment by Rambatino** [comment]
Was going to go package

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+setup(
+    # Application name:
+    name="ComStats",
+
+    # Version number (initial):
+    version="0.1.0",
+
+    # Application author details:
+    author="Zappistore",
+    author_email="mark@zappistore.com",
```
Comment (xulaus) [inline setup.py]: `datascience@zappi.io` maybe instead? No biggie either way.

**Review by xulaus — APPROVED** [review]
Not super convinced the package way is the best way to go. I'm more on the submodule train now I've thought about it. That way we don't polute pypi with a package we don't intend to maintain indefinitely. That said, its your baby.

**Comment by justfish09** [comment]
code looks fine to me. I have reservations about not having some sort of continuous integration testing before merging though. I would think it should be a requirement instead of `hey, specs pass locally`.

**Comment by Rambatino** [comment]
@justfish09 fine I'll set it up. I think the short answer was i couldn't be bothered. Will do it now!

@xulaus can change the email. Will do some due diligence on the submodule implementation and come back with something concrete

**Comment by Rambatino** [comment]
image

**Comment by Rambatino** [comment]
@xulaus why wouldn't we aim to maintain it indefinitely? Basically it comes down to imports right (as a consequence of folder structure)?

### package
```python
import comstats as cs
```

### submodule
```python
from comstats import stats # or whatever you call the folder and file
```

I prefer the former and given the size of the project I'm not too fussed by modularising it. That said if you feel strongly I can change it.

**Comment by xulaus** [comment]
Re maintaining: My understanding was that this would be a temporary fix and eventualy everything would be merged together in one big repo. If the plan is to keep this seperate then thats my major objection neutralised.

**Pull request merged**